vtex-apps/checkout-ui-custom
# [CHECKOUTUI-130] Fix state selection when short name does not match shortCode in cities mapping

#### What does this PR do? *

Google is returning the short code for Grand Est, France as `Grand Est` instead of `GES`. This PR selects the state by first checking the name returned from Google against the shortCode then against the name from the `_cities` mapping.

#### How to test it? *

Test in workspace [anna](https://anna--sandboxusdev.myvtex.com/checkout/cart/add/?sku=11&qty=1&seller=1&sc=1). Input the address `8 Rue Pierre Perrat` and select the first suggestion presented by Google. The state field should have `Grand Est` pre-selected. Previously, the `Grand Est` option wasn't recognized so the selected state defaulted to the first option in the list or displayed an error message.

<img width="581" alt="Screen Shot 2022-11-10 at 11 46 20 AM" src="https://user-images.githubusercontent.com/53097865/201156111-69aaa362-3aa4-4f65-89fc-45a0e436bb92.png">
